Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -831,6 +833,25 @@ public final class ManifestLoader: ManifestLoaderProtocol {
         // Bin dir will be set when developing swiftpm without building all of the runtimes.
         return resources.binDir ?? resources.libDir.appending(version.runtimeSubpath)
     }
+
+    /// Returns path to the manifest database inside the given cache directory.
+    private static func manifestCacheDBPath(_ cacheDir: AbsolutePath) -> AbsolutePath {
+        return cacheDir.appending(component: "manifest.db")
+    }
+
+    func createCacheIfNeeded() throws {
+        // Return if we have already created the cache.
+        guard self.cache == nil else { return }
+        guard let manifestCacheDBPath = cacheDir.flatMap({ Self.manifestCacheDBPath($0) }) else { return }
+        self.cache = try SQLiteBackedPersistentCache(cacheFilePath: manifestCacheDBPath)
+    }
+
+    public func resetCache() throws {
```

Comment:
Naming nitpick; the semantics here are really to remove the cache, right, and not just reset it?  Or is that considered an implementation detail, i.e. could a different implementation just remove all entries but keep it open?

Reply:
The expectation here is to reset the cache and not remove/disable it. For example, if the cache was just an in-memory structure then the entires would go away on `restCache` but they would get re-populated on subsequent access. `removeCache` sounds as if the client no longer wants to use the cache at all.

Reply:
That makes sense, though it does leave it up to the implementation about whether or not the database will be recreated and opened, i.e. whether it's safe to really remove the directory.  In any case, this is fine.  Thanks!